hera_qm/tests/test_xrfi.py

```diff
@@ -308,145 +308,279 @@ def test_detrend_medfilt(self):
         if os.path.exists(sum_file):
             os.remove(sum_file)
         cmd = ' '.join([arguments, xx_file])
         args = a.parse_args(cmd.split())
-        xrfi.xrfi_run(args.files, args, cmd)
+        xrfi.xrfi_run(args.filename, args, cmd)
         nt.assert_true(os.path.exists(dest_file))
         nt.assert_true(os.path.exists(sum_file))
 
+    def test_xrfi_run_model_and_cal(self):
+        # get argument object
+        a = utils.get_metrics_ArgumentParser('xrfi_run')
+        arg0 = "--xrfi_path={}".format(os.path.join(DATA_PATH, 'test_output'))
+        arg1 = "--algorithm=xrfi_simple"
+        model_file = os.path.join(DATA_PATH, 'zen.2457698.40355.xx.HH.uvc.vis.uvfits')
+        arg2 = "--model_file=" + model_file
+        calfits_file = os.path.join(DATA_PATH, 'zen.2457698.40355.xx.HH.uvcAA.omni.calfits')
+        arg3 = "--calfits_file=" + calfits_file
+        arguments = ' '.join([arg0, arg1, arg2, arg3])
+
+        # test running on our test data
+        xx_file = os.path.join(DATA_PATH, 'zen.2457698.40355.xx.HH.uvcAA')
+        dest_files = []
+        dest_files.append(os.path.join(DATA_PATH, 'test_output',
+                                       'zen.2457698.40355.xx.HH.uvcAA.flags.npz'))
+        dest_files.append(os.path.join(DATA_PATH, 'test_output',
+                                       'zen.2457698.40355.xx.HH.uvc.vis.uvfits.flags.npz'))
+        dest_files.append(os.path.join(DATA_PATH, 'test_output',
+                                       'zen.2457698.40355.xx.HH.uvcAA.omni.calfits.g.flags.npz'))
+        dest_files.append(os.path.join(DATA_PATH, 'test_output',
+                                       'zen.2457698.40355.xx.HH.uvcAA.omni.calfits.x.flags.npz'))
+        for f in dest_files:
+            if os.path.exists(f):
+                os.remove(f)
+        cmd = ' '.join([arguments, xx_file])
+        args = a.parse_args(cmd.split())
+        xrfi.xrfi_run(args.filename, args, cmd)
+        for f in dest_files:
+            nt.assert_true(os.path.exists(f))
+
+    def test_xrfi_run_model_and_cal_errors(self):
+        from pyuvdata import UVData
+
+        # get argument object
+        a = utils.get_metrics_ArgumentParser('xrfi_run')
+        arg0 = "--xrfi_path={}".format(os.path.join(DATA_PATH, 'test_output'))
+        arg1 = "--algorithm=xrfi_simple"
+        model_file = os.path.join(DATA_PATH, 'zen.2457698.40355.xx.HH.uvc.vis.uvfits')
+        arg2 = "--model_file=" + model_file
+        calfits_file = os.path.join(DATA_PATH, 'zen.2457698.40355.xx.HH.uvcAA.omni.calfits')
+        arg3 = "--calfits_file=" + calfits_file
+        xx_file = os.path.join(DATA_PATH, 'zen.2457698.40355.xx.HH.uvcAA')
+        arguments = ' '.join([arg0, arg1, arg2, arg3])
+
+        arg4 = "--model_file_format=miriad"
+        cmd = ' '.join([arguments, arg4, xx_file])
+        args = a.parse_args(cmd.split())
+        nt.assert_raises(RuntimeError, xrfi.xrfi_run, args.filename, args, cmd)
```

Comment:
I think this is the line that's causing the error in the tests. (Locally, my nosetests crashes hard and won't finish running the rest of the tests.) I think the 3 solutions to this are: 1) We don't support Miriad as a format for the model visibilities, 2) We support Miriad as a format, but remove this test (so we don't cover the `uvm.read_miriad` line), or 3) convert the model visibility file from uvfits to Miriad and use that for the test.

I think for me, 3) is the most appealing solution, and can probably be done on-the-fly as part of the test. Then the Miriad model visibility file can be removed at the end of the test along with the test outputs. I'm open to other solutions too.